### ISSUE TITLE
how to setup cascade delete for relationship?

### ISSUE THREAD

**Issue by raymclee** [romeerez/orchid-orm#4]
Hi, found and trying this orm recently and loving it! I wonder is there a way to setup cascade delete for relationship for now?
Thanks!

**Comment by raymclee** [comment]
sorry just found it in the doc
https://orchid-orm.netlify.app/guide/migration-column-methods.html#foreignkey

**Comment by raymclee** [comment]
but when i try to change the foreignkey with
```js
t.change(
      t.integer(),
      t.integer().foreignKey("user", "id", { onDelete: "CASCADE" })
    ),
```
but it still refusing me to delete

**Comment by romeerez** [comment]
Hi, thank you for trying it!

In the long walkthrough guide I encountered similar issue, I guess it is the one that you have: [https://orchid-orm.netlify.app/guide/building-a-sample-app.html#delete-an-article](url)

There I had to delete article tags explicitly before deleting the article.

This ORM is partially inspired by ActiveRecord from Rails, where you can do so:

````
class Post ...
  has_many :comments, :dependent => :delete_all
end
````

So all comments are deleted automatically when the post is going to be deleted.

It would be nice to have a similar feature here, like this:

````
export class PostModel extends Model {
  table = 'user'
  columns = // ...columns
  
  relations = {
    comments: this.hasMany(() => CommentModel, {
      primaryKey: 'id',
      foreignKey: 'userId',
      onDelete: 'cascade', // Delete comments before the Post
    }),
  }
}
````

But need to put more thought for how to name the new option, what possible values could be, what possible situations.

I'll work on it, thank you for pointing! I'll be grateful if you continue to try it and create more issues!

**Comment by romeerez** [comment]
My previous comment is irrelevant, it all should work with `onDelete: 'CASCADE'` option in the migration

But unfortunately, when changing the column in a migration, the option `onDelete` is not handled, it is a bug and it will be fixed soon.

**Comment by raymclee** [comment]
> My previous comment is irrelevant, it all should work with `onDelete: 'CASCADE'` option in the migration
> 
> But unfortunately, when changing the column in a migration, the option `onDelete` is not handled, it is a bug and it will be fixed soon.

Cool thanks.
Does it work when creating table in migration?

**Comment by romeerez** [comment]
Yes, it should work fine, so while it isn't yet fixed for changing the column, you can `npm run db rollback`, update code, and `npm run db migrate` to reapply the migration

**Comment by romeerez** [comment]
The fix for this is ready, it is published in `rake-db` 2.1.2 version.

Added some additional docs [here](https://orchid-orm.netlify.app/guide/migration-writing.html#change) for changing foreign keys.

Along with foreign keys, it is now possible to change indexes in the same manner.
